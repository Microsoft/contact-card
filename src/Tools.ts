import * as React from "react";
import { Button, BaseButton, LinkBase } from "office-ui-fabric-react";

export function openLink(url: string, e?: React.SyntheticEvent<HTMLAnchorElement | HTMLButtonElement | HTMLElement | HTMLDivElement | BaseButton | Button | LinkBase>) {
    let target: string | undefined = "_self";   // open in same window by default. This also prevents empty tabs for `mailto:` links in Chrome

    if (window.self !== window.top) {           // CSP (Content Security Policy) can prevent URL changing if we're inside iFrame. We let browser to decide where to open the URL
        target = undefined;
    }

    window.open(url, target);

    if (e) {
        e.preventDefault();
        e.stopPropagation();
    }
}
