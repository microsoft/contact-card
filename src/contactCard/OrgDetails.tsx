import * as React from "react";
import { css, ActionButton, AnimationClassNames, Label, PersonaSize, Shimmer, ShimmerElementType } from "@fluentui/react";
import { Persona } from "../Persona";
import { PersonaShowMode, IPersonaProfile } from "../Types";


export function renderOrgHierarchy(profile: IPersonaProfile, allManagers: IPersonaProfile[] | undefined, directs: IPersonaProfile[] | undefined, onPersonaClick: (profile: IPersonaProfile) => void): React.ReactNode {
    return allManagers ?
        (
            <>
                <Label className="org-hierarchy-title">Organization</Label>
                <ul className={css("org-hierarchy", AnimationClassNames.slideLeftIn400)}>
                    {renderAllManagers(allManagers, onPersonaClick)}
                    <li className="person-current">
                        <Persona
                            id={profile.id}
                            displayName={profile.displayName}
                            showMode={PersonaShowMode.NameTitleDepartment}
                            size={PersonaSize.size48}
                        />
                    </li>
                    {renderDirects(directs, onPersonaClick)}
                </ul>
            </>
        ) :
        (
            <ul className={css("org-hierarchy", AnimationClassNames.slideLeftIn400)}>
                <li className="person"><Shimmer shimmerElements={[{ type: ShimmerElementType.circle, height: 48 }, { type: ShimmerElementType.gap, width: 12 }, { type: ShimmerElementType.line }]} width={"80%"} /></li>
                <li className="hierarchy-splitter" />
                <li className="person"><Shimmer shimmerElements={[{ type: ShimmerElementType.circle, height: 48 }, { type: ShimmerElementType.gap, width: 12 }, { type: ShimmerElementType.line }]} width={"80%"} /></li>
                <li className="hierarchy-splitter" />
                <li className="person"><Shimmer shimmerElements={[{ type: ShimmerElementType.circle, height: 48 }, { type: ShimmerElementType.gap, width: 12 }, { type: ShimmerElementType.line }]} width={"80%"} /></li>
            </ul>
        );
}


function renderAllManagers(allManagers: IPersonaProfile[], onPersonaClick: (profile: IPersonaProfile) => void): React.ReactNode[] {
    const res: React.ReactNode[] = [];

    for (let i = allManagers.length - 1; i >= 0; --i) {
        const mngr = allManagers[i];
        res.push(
            <li key={i}>
                <ActionButton className="person manager" onClick={() => onPersonaClick(mngr)}>
                    <Persona
                        id={mngr.id}
                        displayName={mngr.displayName}
                        showMode={PersonaShowMode.NameTitleDepartment}
                        size={PersonaSize.size48}
                    />
                </ActionButton>
            </li>
        );
        res.push(<li className="hierarchy-splitter" key={`spl-${i}`} />);
    }

    return res;
}


function renderDirects(directs: IPersonaProfile[] | undefined, onPersonaClick: (profile: IPersonaProfile) => void): React.ReactNode {
    if (!directs || !directs.length) {
        return <></>;
    }

    let key = 0;
    return (
        <>
            <li className="hierarchy-splitter" />
            <li className="directs">
                <Label className="directs-label">Direct reports ({directs.length})</Label>
                <ul>
                    {directs.map(direct => (
                        <li key={key++}>
                            <ActionButton className="person direct" onClick={() => onPersonaClick(direct)}>
                                <Persona
                                    id={direct.id}
                                    displayName={direct.displayName}
                                    showMode={PersonaShowMode.NameTitle}
                                    size={PersonaSize.size48}
                                />
                            </ActionButton>
                        </li>
                    ))}
                </ul>
            </li>
        </>
    );
}
