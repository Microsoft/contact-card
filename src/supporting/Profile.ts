import { IPersonaProfile } from "../Types";


export function buildProfileResponse(id: number) {
    return {
        id: `userId${id}`,
        displayName: `Display Name ${id}`,
        jobTitle: `Job Title ${id}`,
        department: `Department ${id}`,
        mail: `user${id}@contoso.com`,
        officeLocation: `1234-${id}`,
        city: "Redmond",
        businessPhones: [`555-123-456${id}`],
        imAddresses: [`im-user${id}@contoso.com`],
        companyName: "Contoso"
    };
}


export function buildProfile(id: number): IPersonaProfile {
    return {
        id: `userId${id}`,
        displayName: `Display Name ${id}`,
        jobTitle: `Job Title ${id}`,
        department: `Department ${id}`,
        email: `user${id}@contoso.com`,
        officeLocation: `1234-${id}`,
        city: "Redmond",
        businessPhone: `555-123-456${id}`,
        imAddress: `im-user${id}@contoso.com`,
        companyName: "Contoso"
    };
}
