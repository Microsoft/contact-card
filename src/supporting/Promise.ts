export async function delay(ms: number) {
    return new Promise((resolve) => setTimeout(resolve, ms));
}

export async function waitNextTick() {
    return delay(0);
}
