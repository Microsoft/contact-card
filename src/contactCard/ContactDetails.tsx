import * as React from "react";
import { css, AnimationClassNames, Label, Link } from "office-ui-fabric-react";
import { IPersonaProfile } from "../Types";
import { openLink } from "../Tools";


export function renderContactDetails(profile: IPersonaProfile): React.ReactNode {
    return (
        <ul className={css("contact-details", AnimationClassNames.slideLeftIn400)}>
            <li>
                <Label>Email</Label>
                <Link href={`mailto:${profile.email}`} className="contact-link email" onClick={e => openLink(`mailto:${profile.email}`, e)}>{profile.email}</Link>
            </li>
            {profile.imAddress && (
                <li>
                    <Label>Chat</Label>
                    <Link href={`sip:${profile.imAddress}`} className="contact-link chat" onClick={e => openLink(`sip:${profile.imAddress}`, e)}>{profile.imAddress}</Link>
                </li>
            )}
            <li>
                <Label>Work phone</Label>
                <Link href={`tel:${profile.businessPhone}`} className="contact-link business-phone" onClick={e => openLink(`tel:${profile.businessPhone}`, e)}>{profile.businessPhone}</Link>
            </li>
            <li>
                <Label>Company</Label>
                <span>{profile.companyName}</span>
            </li>
            <li>
                <Label>Job title</Label>
                <span>{profile.jobTitle}</span>
            </li>
            <li>
                <Label>Department</Label>
                <span>{profile.department}</span>
            </li>
            <li>
                <Label>Office location</Label>
                <span>{profile.officeLocation}</span>
                <span>&nbsp;{profile.city}</span>
            </li>
        </ul>
    );
}
