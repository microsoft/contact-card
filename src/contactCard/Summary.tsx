import * as React from "react";
import { ActionButton, Icon, Link, PersonaSize, Shimmer, ShimmerElementType } from "@fluentui/react";
import { Persona } from "../Persona";
import { IPersonaProfile, PersonaShowMode } from "../Types";
import { openLink } from "../Tools";


export function renderSummary(
    profile: IPersonaProfile,
    manager: IPersonaProfile | undefined,
    isManagerLoading: boolean,
    onContactDetailsClick: () => void,
    onOrgDetailsClick: () => void,
    onPersonaClick: (profile: IPersonaProfile) => void
): React.ReactNode {
    return (
        <ul tabIndex={-1} className="summary">
            {renderContactSummary(profile, onContactDetailsClick)}
            {renderOrgSummary(manager, isManagerLoading, onOrgDetailsClick, onPersonaClick)}
        </ul>
    ); 
}

    let isExpanded = false;
    function handleContactDetailsClick(onContactDetailsClick: () => void) {
        onContactDetailsClick();
        const button = document.querySelector('[data-focus="button"]');
        if (button) {
          (button as HTMLButtonElement).focus();
        }
        isExpanded = true;
    }
  
function renderContactSummary(profile: IPersonaProfile, onContactDetailsClick: () => void): React.ReactNode {
    return (
        <li>
            <ActionButton className="section-title contact-details-button" onClick={onContactDetailsClick}>
                Contact <Icon iconName="ChevronRight" className="chevron-icon" />
            </ActionButton>

            <div className="contact-row">
                <Icon iconName="Mail" className="contact-icon" />
                <Link href={`mailto:${profile.email}`} className="contact-link email" onClick={e => openLink(`mailto:${profile.email}`, e)} aria-label={`Email ${profile.email}`}>{profile.email}</Link>
            </div>
            {
                profile.businessPhone &&
                <div className="contact-row">
                    <Icon iconName="Phone" className="contact-icon" />
                    <Link href={`tel:${profile.businessPhone}`} className="contact-link business-phone" onClick={e => openLink(`tel:${profile.businessPhone}`, e)} aria-label={`Phone ${profile.businessPhone}`}>{profile.businessPhone}</Link>
                </div>
            }
            <div className="contact-row">
                <Icon iconName="POI" className="contact-icon" />
                <span>{profile.officeLocation}</span>
                <span>&nbsp;{profile.city}</span>
            </div>
            <ActionButton className="more-details contact-details" onClick={() => handleContactDetailsClick(onContactDetailsClick)} aria-label={isExpanded ? 'show more button expanded' : 'show more button collapsed'} aria-expanded={isExpanded} data-focus="button" aria-live="polite">
            {isExpanded ? "expanded" : "collapsed"}
                Show more
            </ActionButton>
        </li>
    );
}


function renderOrgSummary(
    manager: IPersonaProfile | undefined,
    isManagerLoading: boolean,
    onOrgDetailsClick: () => void,
    onPersonaClick: (profile: IPersonaProfile) => void
): React.ReactNode {
    return (
        <li>
            {(manager || isManagerLoading) &&
                <>
                    <ActionButton className="section-title org-details-button" onClick={onOrgDetailsClick}>
                        Reports to <Icon iconName="ChevronRight" className="chevron-icon" />
                    </ActionButton>
                    {
                        manager ?
                            <ActionButton className="person manager" onClick={() => onPersonaClick(manager)}>
                                <Persona
                                    id={manager.id}
                                    displayName={manager.displayName}
                                    showMode={PersonaShowMode.NameTitle}
                                    size={PersonaSize.size40}
                                />
                            </ActionButton>
                            :
                            <div className="person">
                                <Shimmer shimmerElements={[{ type: ShimmerElementType.circle, height: 40 }, { type: ShimmerElementType.gap, width: 12 }, { type: ShimmerElementType.line }]} width={"80%"} />
                            </div>
                    }
                </>
            }
            <ActionButton className="more-details org-details" onClick={onOrgDetailsClick}>
                Show organization
            </ActionButton>
        </li>
    );
}
