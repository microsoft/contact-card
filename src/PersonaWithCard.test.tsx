import * as React from "react";
import * as Enzyme from "enzyme";
import { buildProfile } from "./supporting/Profile";
import { PersonaShowMode, IPersonaProfile } from "./Types";
import { GraphService } from "./GraphService";
import { waitNextTick } from "./supporting/Promise";
import { PersonaSize, HoverCard } from "@fluentui/react";
import { PersonaWithCard } from "./PersonaWithCard";
import { renderContactDetails } from "./contactCard/ContactDetails";
import { renderOrgHierarchy } from "./contactCard/OrgDetails";
import { renderSummary } from "./contactCard/Summary";
import { openLink } from "./Tools";

jest.mock("./Tools");
jest.mock("./GraphService");
jest.mock("./contactCard/ContactDetails");
jest.mock("./contactCard/OrgDetails");
jest.mock("./contactCard/Summary");

// tslint:disable:no-non-null-assertion

beforeEach(() => {
    // we've tested this components separately, we can mock them now
    (renderContactDetails as jest.Mock).mockImplementation((...args) => <div {...args} id="ContactDetailsMock" />);
    (renderOrgHierarchy as jest.Mock).mockImplementation((...args) => <div {...args} id="OrgHierarchyMock" />);
    (renderSummary as jest.Mock).mockImplementation((...args) => <div {...args} id="SummaryMock" />);
});


afterEach(() => {
    jest.resetAllMocks();
});


test("renders PersonaWithCard initial state", async () => {
    const wrap = Enzyme.shallow(<PersonaWithCard id="userId1" showMode={PersonaShowMode.NameOnly} />);
    expect(wrap).toMatchSnapshot();
});


test("renders PersonaWithCard with extra props", async () => {
    const wrap = Enzyme.shallow(<PersonaWithCard id="userId1" displayName="Initial Display Name 1" size={PersonaSize.size100} showMode={PersonaShowMode.NameOnly} />);
    expect(wrap).toMatchSnapshot();
});


test("renders PersonaWithCard fails if no Id and EMail", async () => {
    expect(() => Enzyme.shallow(<PersonaWithCard showMode={PersonaShowMode.NameOnly} />))
        .toThrowError(/required/);
});


test("re-renders PersonaWithCard by Email", async () => {
    const wrap = Enzyme.shallow(<PersonaWithCard email="userId1@contoso.com" showMode={PersonaShowMode.NameOnly} />);
    const render1 = wrap.html();

    wrap.setProps({ email: "userId2@contoso.com", showMode: PersonaShowMode.NameOnly });
    const render2 = wrap.html();

    expect(render1).not.toBe(render2);
});


test("renders Compact card", async () => {
    const compactCard = renderCompactCard(await renderOpenedCard());
    expect(compactCard).toMatchSnapshot();
});


test("renders Compact card w/o IM address", async () => {
    const profile = buildProfile(1);
    profile.imAddress = undefined;
    const compactCard = renderCompactCard(await renderOpenedCard(profile));
    expect(compactCard).toMatchSnapshot();
});


test("opens link in Compact card", async () => {
    const compactCard = renderCompactCard(await renderOpenedCard());

    compactCard.find(".mail-link").simulate("click");
    expect(openLink).toBeCalledWith(`mailto:${buildProfile(1).email}`, undefined);

    compactCard.find(".chat-link").simulate("click");
    expect(openLink).toBeCalledWith(`sip:${buildProfile(1).imAddress}`, undefined);
});


test("renders Expanded card with progress", async () => {
    GraphService.resolveProfile = async () => Promise.resolve(buildProfile(1));
    GraphService.getManager = async () => Promise.resolve(buildProfile(2));

    const openedCard = Enzyme.shallow(<PersonaWithCard id="userId1" showMode={PersonaShowMode.NameOnly} />);
    openedCard.find(HoverCard).props().onCardVisible();

    const expandedCard = renderExpandedCard(openedCard);
    expect(expandedCard).toMatchSnapshot();
});


test("renders Expanded card with Summary", async () => {
    const expandedCard = renderExpandedCard(await renderOpenedCard());
    expect(expandedCard).toMatchSnapshot();
});


test("renders Expanded card with ContactDetails", async () => {
    const openedCard = await renderOpenedCard();
    let expandedCard = renderExpandedCard(openedCard);

    (renderSummary as jest.Mock).mock.calls[0][3]();        // call showContactDetails()
    expandedCard = renderExpandedCard(openedCard);
    expect(expandedCard).toMatchSnapshot();
});


test("renders Compact card with ContactDetails", async () => {
    const openedCard = await renderOpenedCard();
    renderExpandedCard(openedCard);
    (renderSummary as jest.Mock).mock.calls[0][3]();        // call showContactDetails()

    const compactCard = renderCompactCard(openedCard);
    expect(compactCard).toMatchSnapshot();
});


test("switches to summaryMode when clicking on person in Compact card", async () => {
    const openedCard = await renderOpenedCard();
    renderExpandedCard(openedCard);
    (renderSummary as jest.Mock).mock.calls[0][3]();        // call showContactDetails()
    const compactCard = renderCompactCard(openedCard);
    compactCard.find(".person").simulate("click");

    const expandedCard = renderExpandedCard(openedCard);
    expect(expandedCard).toMatchSnapshot();
});


test("renders Expanded card with OrgHierarchy", async () => {
    GraphService.getAllManagers = async () => Promise.resolve([buildProfile(2), buildProfile(3)]);
    GraphService.getDirects = async () => Promise.resolve([buildProfile(12), buildProfile(13)]);

    const openedCard = await renderOpenedCard();
    let expandedCard = renderExpandedCard(openedCard);

    (renderSummary as jest.Mock).mock.calls[0][4]();        // call showOrgHierarchy()
    await waitNextTick();
    expandedCard = renderExpandedCard(openedCard);
    expect(expandedCard).toMatchSnapshot();
});


test("renders next Person", async () => {
    const openedCard = await renderOpenedCard();
    renderExpandedCard(openedCard);

    (renderSummary as jest.Mock).mock.calls[0][5](buildProfile(2));        // call showNextPerson()
    await waitNextTick();

    const compactCard = renderCompactCard(openedCard);
    expect(compactCard).toMatchSnapshot();
});


test("renders prev Person", async () => {
    const openedCard = await renderOpenedCard();
    renderExpandedCard(openedCard);

    (renderSummary as jest.Mock).mock.calls[0][5](buildProfile(2));        // call showNextPerson()
    await waitNextTick();
    let compactCard = renderCompactCard(openedCard);
    compactCard.find(".back-button").simulate("click");
    await waitNextTick();

    compactCard = renderCompactCard(openedCard);
    expect(compactCard).toMatchSnapshot();
});


async function renderOpenedCard(profile?: IPersonaProfile) {
    const extractNum = (s: string) => parseInt(s.match(/\d+/)![0], 10);
    GraphService.resolveProfile = async (userId) => Promise.resolve(profile || buildProfile(extractNum(userId)));
    GraphService.getManager = async (userId) => Promise.resolve(buildProfile(extractNum(userId) + 1));

    const wrap = Enzyme.shallow(<PersonaWithCard id="userId1" showMode={PersonaShowMode.NameOnly} />);
    wrap.find(HoverCard).props().onCardVisible();

    await waitNextTick();

    return wrap;
}


function renderCompactCard(openedCard: Enzyme.ShallowWrapper) {
    return Enzyme.shallow(<div>{openedCard.find(HoverCard).props().expandingCardProps.onRenderCompactCard()}</div>);
}


function renderExpandedCard(openedCard: Enzyme.ShallowWrapper) {
    return Enzyme.shallow(<div>{openedCard.find(HoverCard).props().expandingCardProps.onRenderExpandedCard()}</div>);
}
