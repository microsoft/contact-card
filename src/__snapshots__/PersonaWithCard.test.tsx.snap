// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Compact card 1`] = `
<div>
  <div
    className="compact-card"
  >
    <Persona
      displayName="Display Name 1"
      email="user1@contoso.com"
      id="userId1"
      showMode={7}
      size={14}
    />
    <FocusZone
      direction={1}
      handleTabKey={1}
      isCircularNavigation={false}
      shouldRaiseClicks={true}
    >
      <CustomizedActionButton
        className="mail-link"
        iconProps={
          Object {
            "iconName": "Mail",
          }
        }
        onClick={[Function]}
      >
        Send email
      </CustomizedActionButton>
      <CustomizedActionButton
        className="chat-link"
        iconProps={
          Object {
            "iconName": "Chat",
          }
        }
        onClick={[Function]}
      >
        Chat
      </CustomizedActionButton>
    </FocusZone>
  </div>
</div>
`;

exports[`renders Compact card w/o IM address 1`] = `
<div>
  <div
    className="compact-card"
  >
    <Persona
      displayName="Display Name 1"
      email="user1@contoso.com"
      id="userId1"
      showMode={7}
      size={14}
    />
    <FocusZone
      direction={1}
      handleTabKey={1}
      isCircularNavigation={false}
      shouldRaiseClicks={true}
    >
      <CustomizedActionButton
        className="mail-link"
        iconProps={
          Object {
            "iconName": "Mail",
          }
        }
        onClick={[Function]}
      >
        Send email
      </CustomizedActionButton>
    </FocusZone>
  </div>
</div>
`;

exports[`renders Compact card with ContactDetails 1`] = `
<div>
  <div
    className="compact-card css-165"
  >
    <CustomizedActionButton
      className="person"
      onClick={[Function]}
    >
      <Persona
        displayName="Display Name 1"
        email="user1@contoso.com"
        id="userId1"
        showMode={1}
        size={12}
      />
    </CustomizedActionButton>
  </div>
</div>
`;

exports[`renders Expanded card with ContactDetails 1`] = `
<div>
  <div
    className="extended-card"
  >
    <div
      0={
        Object {
          "businessPhone": "555-123-4561",
          "city": "Redmond",
          "companyName": "Contoso",
          "department": "Department 1",
          "displayName": "Display Name 1",
          "email": "user1@contoso.com",
          "id": "userId1",
          "imAddress": "im-user1@contoso.com",
          "jobTitle": "Job Title 1",
          "officeLocation": "1234-1",
        }
      }
      id="ContactDetailsMock"
    />
  </div>
</div>
`;

exports[`renders Expanded card with OrgHierarchy 1`] = `
<div>
  <div
    className="extended-card"
  >
    <div
      0={
        Object {
          "businessPhone": "555-123-4561",
          "city": "Redmond",
          "companyName": "Contoso",
          "department": "Department 1",
          "displayName": "Display Name 1",
          "email": "user1@contoso.com",
          "id": "userId1",
          "imAddress": "im-user1@contoso.com",
          "jobTitle": "Job Title 1",
          "officeLocation": "1234-1",
        }
      }
      1={
        Array [
          Object {
            "businessPhone": "555-123-4562",
            "city": "Redmond",
            "companyName": "Contoso",
            "department": "Department 2",
            "displayName": "Display Name 2",
            "email": "user2@contoso.com",
            "id": "userId2",
            "imAddress": "im-user2@contoso.com",
            "jobTitle": "Job Title 2",
            "officeLocation": "1234-2",
          },
          Object {
            "businessPhone": "555-123-4563",
            "city": "Redmond",
            "companyName": "Contoso",
            "department": "Department 3",
            "displayName": "Display Name 3",
            "email": "user3@contoso.com",
            "id": "userId3",
            "imAddress": "im-user3@contoso.com",
            "jobTitle": "Job Title 3",
            "officeLocation": "1234-3",
          },
        ]
      }
      2={
        Array [
          Object {
            "businessPhone": "555-123-45612",
            "city": "Redmond",
            "companyName": "Contoso",
            "department": "Department 12",
            "displayName": "Display Name 12",
            "email": "user12@contoso.com",
            "id": "userId12",
            "imAddress": "im-user12@contoso.com",
            "jobTitle": "Job Title 12",
            "officeLocation": "1234-12",
          },
          Object {
            "businessPhone": "555-123-45613",
            "city": "Redmond",
            "companyName": "Contoso",
            "department": "Department 13",
            "displayName": "Display Name 13",
            "email": "user13@contoso.com",
            "id": "userId13",
            "imAddress": "im-user13@contoso.com",
            "jobTitle": "Job Title 13",
            "officeLocation": "1234-13",
          },
        ]
      }
      3={[Function]}
      id="OrgHierarchyMock"
    />
  </div>
</div>
`;

exports[`renders Expanded card with Summary 1`] = `
<div>
  <div
    className="extended-card"
  >
    <div
      0={
        Object {
          "businessPhone": "555-123-4561",
          "city": "Redmond",
          "companyName": "Contoso",
          "department": "Department 1",
          "displayName": "Display Name 1",
          "email": "user1@contoso.com",
          "id": "userId1",
          "imAddress": "im-user1@contoso.com",
          "jobTitle": "Job Title 1",
          "officeLocation": "1234-1",
        }
      }
      1={
        Object {
          "businessPhone": "555-123-4562",
          "city": "Redmond",
          "companyName": "Contoso",
          "department": "Department 2",
          "displayName": "Display Name 2",
          "email": "user2@contoso.com",
          "id": "userId2",
          "imAddress": "im-user2@contoso.com",
          "jobTitle": "Job Title 2",
          "officeLocation": "1234-2",
        }
      }
      2={false}
      3={[Function]}
      4={[Function]}
      5={[Function]}
      id="SummaryMock"
    />
  </div>
</div>
`;

exports[`renders Expanded card with progress 1`] = `
<div>
  <div
    className="extended-card"
  >
    <StyledShimmerBase
      className="shimmer"
    />
    <StyledShimmerBase
      className="shimmer"
    />
    <StyledShimmerBase
      className="shimmer"
    />
  </div>
</div>
`;

exports[`renders PersonaWithCard initial state 1`] = `
<div
  data-is-focusable="true"
  tabIndex={0}
>
  <StyledHoverCardBase
    className="persona-hover-card"
    expandingCardProps={
      Object {
        "onRenderCompactCard": [Function],
        "onRenderExpandedCard": [Function],
      }
    }
    instantOpenOnClick={true}
    onCardVisible={[Function]}
    openHotKey={13}
    setInitialFocus={true}
    target={null}
    trapFocus={true}
  >
    <Persona
      className="contact-card-persona"
      id="userId1"
      showMode={1}
    />
  </StyledHoverCardBase>
</div>
`;

exports[`renders PersonaWithCard with extra props 1`] = `
<div
  data-is-focusable="true"
  tabIndex={0}
>
  <StyledHoverCardBase
    className="persona-hover-card"
    expandingCardProps={
      Object {
        "onRenderCompactCard": [Function],
        "onRenderExpandedCard": [Function],
      }
    }
    instantOpenOnClick={true}
    onCardVisible={[Function]}
    openHotKey={13}
    setInitialFocus={true}
    target={null}
    trapFocus={true}
  >
    <Persona
      className="contact-card-persona"
      displayName="Initial Display Name 1"
      id="userId1"
      showMode={1}
      size={15}
    />
  </StyledHoverCardBase>
</div>
`;

exports[`renders next Person 1`] = `
<div>
  <div
    className="back-section"
  >
    <StyledTooltipHostBase
      content="Back to Display Name 1"
    >
      <CustomizedActionButton
        className="back-button"
        iconProps={
          Object {
            "iconName": "Back",
          }
        }
        onClick={[Function]}
      />
    </StyledTooltipHostBase>
  </div>
  <div
    className="compact-card"
  >
    <Persona
      displayName="Display Name 2"
      email="user2@contoso.com"
      id="userId2"
      showMode={7}
      size={14}
    />
    <FocusZone
      direction={1}
      handleTabKey={1}
      isCircularNavigation={false}
      shouldRaiseClicks={true}
    >
      <CustomizedActionButton
        className="mail-link"
        iconProps={
          Object {
            "iconName": "Mail",
          }
        }
        onClick={[Function]}
      >
        Send email
      </CustomizedActionButton>
      <CustomizedActionButton
        className="chat-link"
        iconProps={
          Object {
            "iconName": "Chat",
          }
        }
        onClick={[Function]}
      >
        Chat
      </CustomizedActionButton>
    </FocusZone>
  </div>
</div>
`;

exports[`renders prev Person 1`] = `
<div>
  <div
    className="compact-card"
  >
    <Persona
      displayName="Display Name 1"
      email="user1@contoso.com"
      id="userId1"
      showMode={7}
      size={14}
    />
    <FocusZone
      direction={1}
      handleTabKey={1}
      isCircularNavigation={false}
      shouldRaiseClicks={true}
    >
      <CustomizedActionButton
        className="mail-link"
        iconProps={
          Object {
            "iconName": "Mail",
          }
        }
        onClick={[Function]}
      >
        Send email
      </CustomizedActionButton>
      <CustomizedActionButton
        className="chat-link"
        iconProps={
          Object {
            "iconName": "Chat",
          }
        }
        onClick={[Function]}
      >
        Chat
      </CustomizedActionButton>
    </FocusZone>
  </div>
</div>
`;

exports[`switches to summaryMode when clicking on person in Compact card 1`] = `
<div>
  <div
    className="extended-card"
  >
    <div
      0={
        Object {
          "businessPhone": "555-123-4561",
          "city": "Redmond",
          "companyName": "Contoso",
          "department": "Department 1",
          "displayName": "Display Name 1",
          "email": "user1@contoso.com",
          "id": "userId1",
          "imAddress": "im-user1@contoso.com",
          "jobTitle": "Job Title 1",
          "officeLocation": "1234-1",
        }
      }
      1={
        Object {
          "businessPhone": "555-123-4562",
          "city": "Redmond",
          "companyName": "Contoso",
          "department": "Department 2",
          "displayName": "Display Name 2",
          "email": "user2@contoso.com",
          "id": "userId2",
          "imAddress": "im-user2@contoso.com",
          "jobTitle": "Job Title 2",
          "officeLocation": "1234-2",
        }
      }
      2={false}
      3={[Function]}
      4={[Function]}
      5={[Function]}
      id="SummaryMock"
    />
  </div>
</div>
`;
